Content.Server/Paper/Plane/PaperPlaneSystem.cs

```diff
@@ -0,0 +1,210 @@
+
+            var plane = EntityManager.AddComponent<PaperPlaneComponent>(paper.Owner);
+
+            if (TryComp<AppearanceComponent>(uid, out var appearance))
+                appearance.SetData(PaperVisuals.Status, PaperStatus.Plane);
+
+            _tagSystem.AddTags(paper.Owner, plane.Tags);
+        }
+
+        private void OnCollideEvent(EntityUid uid, PaperPlaneComponent plane, StartCollideEvent args)
+        {
+            //unlike most thrown items, we want to ignore mobs
+            if ((args.OtherFixture.CollisionLayer & (int) plane.CollisionMask) == 0)
+                return;
+
+            //need to set back to ground on collide for friction to kick in again
+            if (TryComp<PhysicsComponent>(uid, out var physics))
+                physics.BodyStatus = BodyStatus.OnGround;
+        }
+
+        public void OnLand(EntityUid uid, PaperPlaneComponent plane, LandEvent args)
+        {
+            //keep on floatin' baby
+            if (TryComp<PhysicsComponent>(uid, out var physics))
+                physics.BodyStatus = BodyStatus.InAir;
+        }
+
+        private void OnPaperUIOpenEvent(EntityUid uid, PaperPlaneComponent plane, ActivatableUIOpenAttemptEvent args)
+        {
+            //don't open paper UI when folded. User must unfold first
+            args.Cancel();
+        }
+
+        private void OnUnfoldPlaneCancel(EntityUid uid, PaperPlaneComponent paper, UnfoldPlaneCancelledEvent args)
+        {
+            args.Plane.CancelToken = null;
+        }
+
+        private void OnFoldPlaneCancel(EntityUid uid, PaperComponent paper, FoldPlaneCancelledEvent args)
+        {
+            args.Paper.CancelToken = null;
+        }
+
+        public sealed class FoldPlaneEvent : EntityEventArgs
+        {
+            public readonly PaperComponent Paper;
+            public readonly EntityUid User;
+
+            public FoldPlaneEvent(PaperComponent paper, EntityUid user)
+            {
+                Paper = paper;
+                User = user;
+            }
+        }
+
+        public sealed class UnfoldPlaneEvent : EntityEventArgs
+        {
+            public readonly PaperPlaneComponent Plane;
+            public readonly EntityUid User;
+
+            public UnfoldPlaneEvent(PaperPlaneComponent plane, EntityUid user)
+            {
+                Plane = plane;
+                User = user;
+            }
+        }
+
+        public sealed class UnfoldPlaneCancelledEvent : EntityEventArgs
+        {
+            public readonly PaperPlaneComponent Plane;
+
+            public UnfoldPlaneCancelledEvent(PaperPlaneComponent plane)
+            {
+                Plane = plane;
+            }
+        }
+
+        public sealed class FoldPlaneCancelledEvent : EntityEventArgs
+        {
+            public readonly PaperComponent Paper;
+
+            public FoldPlaneCancelledEvent(PaperComponent paper)
+            {
+                Paper = paper;
+            }
+        }
```

Comment:
PaperComponent is unnecessary because you already have component.

Also these can probably just be struct events done by ref.